### PR TITLE
MySQL/PostgreSQL VNET with API Version 2017-12-01 for Public Preview.

### PR DESCRIPTION
**Pull request by rahulgouthamDOTgs**
Note : This API has only version change compared to Previous check in .

Ref PRs :  
Original PR 
https://github.com/Azure/azure-rest-api-specs/pull/1773

Revert due to Private Preview Feature
https://github.com/Azure/azure-rest-api-specs/pull/2411

API version change for Public Preview 
https://github.com/Azure/azure-rest-api-specs-pr/pull/478

@QingqingYuan  , @rohit-joy  FYI .

@marstr  , @sarangan12 for review , Please do not merge until the label is removed  , I will notify proactively . 

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by sarangan12 — APPROVED** [review]

**Comment by sarangan12** [comment]
Waiting for ARM (Gaurav) approval for the new version...

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Why do you have both 202 and 201? Are you using 201 + non-terminal provisioning state + Azure Async operation header for tracking long running operation. Or are you using 202 + location header. the first pattern is recommended for PUT/PATCH

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (ravbhatnagar) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: This should be replaced by the standard "provisioningState" property that ARM recommends to use to indicate the state of the resource during creation.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: This is a long running operation , so we use 202 and location header as of now .

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: for now we are keeping this in sync with Microsoft.SQL Resource Provider for consistency .

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (ravbhatnagar) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Ok. I see other resource types under this provider are also using 202. So this should be fine. Although, please note that 201 is recommended for PUT and PATCH. So something to note for future.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (ravbhatnagar) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Consistency with 100+ providers is what we should be trying to achieve, not with one :). Customers would definitely love that more. Not blocking on it. But please note about such things as it impacts the broader Azure REST API usability, consistency and client experience.

**Comment by ravbhatnagar** [comment]
Signing off!

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Sure .

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json
Comment (rahulgouthamDOTgs) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json]: Thanks , I will look into this soon.

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**
